cpp/src/arrow/type_test.cc

```diff
@@ -379,6 +382,168 @@ TEST(TestFieldPath, Basics) {
                                   FieldPath({s.num_fields() * 2}).Get(s));
 }
 
+TEST(TestFieldPath, GetForTable) {
+  using testing::HasSubstr;
+
+  const int length = 4;
```

Comment:
```suggestion
  constexpr int kLength = 4;
```

Very minor, but `kLength` will make it clear elsewhere it is a constant (I don't really care too much about the `const` -> `constexpr` thing but probably a good habit to get into :)

I'd also consider `kNumRows` instead of `kLength` as it is more obvious to the reader.

Note, this same comment applies in a few places below too.